### PR TITLE
Add OTel.NET Automatic example

### PR DESCRIPTION
**Pull request by pjanotti**
While OTel.NET auto-instr is still in beta it is good to be able to compare it against SFx.NET. This PR allows comparing the two by instrumenting the same application.

The code of the app was moved to a shared location so it is ensured that we are instrumenting the same application.

After this initial PR is merged the OpenTracing code should be removed from the app - avoiding to do it now so the files don't get being treated as new, unnecessarily increasing the size of the PR. 

When OTel release a version supporting MongoDB that should be added to OTel configuration.

Example traces:
 - [SFx](https://app.signalfx.com/#/apm/traces/5100d995c8afa9f868aded7f9d16fc2f)
 - [OTel](https://app.signalfx.com/#/apm/traces/af5fd50608f5b6313776e9434be89443)

### CODE REVIEW

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@aurbiztondo-splunk can you please review the wording in the markdown files? 🙏

File: opentelemetry-tracing/opentelemetry-dotnet/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# OpenTelemetry .NET instrumentation examples
+
+This folder contains sample applications that were instrumented using
```
Comment (aurbiztondo-splunk) [inline opentelemetry-tracing/opentelemetry-dotnet/README.md:3]: I'd remove "that were", and just leave "This folder contains sample applications instrumented using..."

File: opentelemetry-tracing/opentelemetry-dotnet/automatic/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# OpenTelemetry .NET automatic instrumentation examples
+
+This folder contains sample applications that are instrumented using
```
Comment (aurbiztondo-splunk) [inline opentelemetry-tracing/opentelemetry-dotnet/automatic/README.md:3]: Same

File: opentelemetry-tracing/opentelemetry-dotnet/automatic/aspnetcore-and-mongodb/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+source code.
+This example is of a simple
+inventory system that is automatically instrumented by adding
+the configuration of a CLR Profiler and the automatic instrumentation
+to the respective [Dockerfile](./InstrumentContainer/Dockerfile).
+
+## Building and running the example app and client
+
+This multi-container application assumes you have [Docker Compose](https://docs.docker.com/compose/) installed and on your system. To build and run the example services, please clone this repository and from this directory do the following:
+
```
Comment (aurbiztondo-splunk) [inline opentelemetry-tracing/opentelemetry-dotnet/automatic/aspnetcore-and-mongodb/README.md:18]: "and from there do the following". And no need to use "please".

File: signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/README.md [hunk 4]
```diff
@@ -2,15 +2,23 @@
 
 This is an example of automatically producing distributed traces using the
 [SignalFx Tracing Library for .NET](https://github.com/signalfx/signalfx-dotnet-tracing).
-Please examine the instrumented [HttpClient](./src/ExampleClient/Program.cs) and [ASP.NET Core application](./src/AspNetCoreExample/Services/ItemService.cs)
-for custom instrumentation patterns using the OpenTracing API. This example is of a simple
-inventory system that is auto-instrumented via [configuration of the CLR Profiler and tracing library](./src/AspNetCoreExample/Dockerfile).
+Please examine the
+[HttpClient](../../../shared/applications/dotnet/aspnetcore-and-mongodb/src/ClientExample/Program.cs)
+and
+[ASP.NET Core application](../../../shared/applications/dotnet/aspnetcore-and-mongodb/src/AspNetCoreExample/Services/ItemService.cs)
```
Comment (aurbiztondo-splunk) [inline signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/README.md:8]: See above.

**Comment by aurbiztondo-splunk** [comment]
> @aurbiztondo-splunk can you please review the wording in the markdown files? 🙏

Left a few comments!

**Comment by pjanotti** [comment]
Great feedback @aurbiztondo-splunk! Fixed via 0f4e679 and d194f55

**Comment by pjanotti** [comment]
Well observed @pellared it seems that the context propagation from ActivitySource to OpenTracing is broken, see https://app.signalfx.com/#/apm/traces/da9e8721b6757dd7fe1128a3fdc0c713

Later I will be removing the OpenTracing instrumentation as stated before but we should investigate this on the OTel side.

**Pull request merged**